### PR TITLE
Adding an option --ty-spz ALL and adding the specializer to the tests

### PR DESCRIPTION
**Pull request by GuillaumeGen**

### CODE REVIEW

File: src/Pirouette/Specializer.hs [hunk 1]
```diff
@@ -6,9 +6,10 @@ module Pirouette.Specializer
...
 import Pirouette.Specializer.Basics
 
-allSpz :: String -> TypeSpecializer
-allSpz "Bool"   = boolSpz
-allSpz "List"   = listSpz
-allSpz "Unit"   = unitSpz
-allSpz "Tuple2" = tuple2Spz
-allSpz "Maybe"  = maybeSpz
\ No newline at end of file
+allSpz :: String -> Maybe TypeSpecializer
```
Comment (VictorCMiraldo) [inline src/Pirouette/Specializer.hs:9]: Thank you! I was about to send a similar patch!

File: src/Pirouette/Term/ToTla.hs [hunk 2]
```diff
@@ -34,6 +34,7 @@ import           Control.Monad.Reader
 import           Control.Monad.Except
 import           Control.Monad.State
 import           Control.Arrow (first, second, (***))
+import           Control.Exception ( catch )
```
Comment (VictorCMiraldo) [inline src/Pirouette/Term/ToTla.hs:37]: Is this really needed? you added the import but didn't use the imported function in the change to `mkTLATySpecializer`

```suggestion
```

**Review by VictorCMiraldo — APPROVED** [review]
ty

File: src/Pirouette/Term/ToTla.hs (hunk 2 above)
Comment (GuillaumeGen) [reply src/Pirouette/Term/ToTla.hs:37]: Yes, you are right, I added it and then changed my mind and did not use it. It should be suppressed.

**Pull request merged**